examples/menubar/js/menubar-editor.js

```diff
@@ -0,0 +1,810 @@
+/*
+*   This content is licensed according to the W3C Software License at
+*   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+*
+*   File:   menubar-editor.js
+*
+*   Desc:   Creates a menubar to control the styling of text in a textarea element
+*/
+
+var MenubarEditor = function (domNode, actionManager) {
+
+  this.domNode = domNode;
+  this.actionManager = actionManager;
+
+  this.menuitemGroups = {};
+  this.menuOrientation = {};
+  this.isPopup = {};
+  this.openPopups = false;
+
+  this.firstChars = {}; // see Menubar init method
+  this.firstMenuitem = {}; // see Menubar init method
+  this.lastMenuitem = {}; // see Menubar init method
+
+  this.initMenu(domNode)
+  domNode.addEventListener('focusin', this.handleMenubarFocusin.bind(this));
+  domNode.addEventListener('focusout', this.handleMenubarFocusout.bind(this));
+
+  window.addEventListener('mousedown', this.handleBackgroundMousedown.bind(this), true);
+};
+
+MenubarEditor.prototype.getMenuitems = function(domNode) {
+  var nodes = [];
+
+  var initMenu = this.initMenu.bind(this);
+  var getGroupId = this.getGroupId.bind(this);
+  var menuitemGroups = this.menuitemGroups;
+
+  var handleMenuMouseover =  this.handleMenuMouseover.bind(this);
+  var handleMenuMouseout  = this.handleMenuMouseout.bind(this);
+
+  function findMenuitems(node, group) {
+    var role, flag, groupId;
+
+    while (node) {
+      flag = true;
+      role = node.getAttribute('role');
+
+      if (role) {
+        role = role.trim().toLowerCase();
```

Comment:
I think both `trim()` and `toLowerCase()` are technically incorrect to normalize the `role` attribute, but the spec isn't very clear: https://github.com/w3c/aria/issues/1236

I think this entire `if` block can be removed.